.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,50 @@
+name: Run Benchmarks
+
+# on:
+#   schedule:
+#     - cron: '0 0 * * 0'  # Runs weekly on Sunday at midnight
```

Comment:
Are we planning to enable this regularly?

Reply:
Yes, I was planning to run this pipeline only as a cron job, so I added the PR trigger as to test if the pipeline's running fine. But it seems like we can run this for every PR and periodically.